### PR TITLE
CLOUDSTACK: Add option to expunge vm on destroy

### PR DESCRIPTION
**Pull request by boul**

### CODE REVIEW

File: libcloud/compute/drivers/cloudstack.py [hunk 1]
```diff
@@ -1034,15 +1034,27 @@ def _create_args_to_params(self, node, **kwargs):
...
         return server_params
 
-    def destroy_node(self, node):
+    def destroy_node(self, node, ex_expunge=None):
         """
         @inherits: :class:`NodeDriver.reboot_node`
         :type node: :class:`CloudStackNode`
 
+        :keyword    ex_expunge: If true is passed, the vm is expunged
+                                immediately. False by default.
```
Comment (sebgoa) [inline libcloud/compute/drivers/cloudstack.py:1043]: might be better to set it to False by default then, and check if it's True before passing it.

Comment (Kami) [inline libcloud/compute/drivers/cloudstack.py:1043]: Yeah, the docstring even says it defaults to `False`.

**Comment by boul** [comment]
Check, changed the default to be False and only passing the arg when True as Cloudstack defaults to False so there is no need to send False to Cloudstack, this way it should also work with older CS versions which do not expose this feature.

File: libcloud/compute/drivers/cloudstack.py [hunk 2]
```diff
@@ -1034,15 +1034,27 @@ def _create_args_to_params(self, node, **kwargs):
...
+        :type       ex_expunge: ``bool``
+
         :rtype: ``bool``
         """
+
+        args = {
+            'id': node.id,
+        }
+
+        if ex_expunge is not False:
```
Comment (sebgoa) [inline libcloud/compute/drivers/cloudstack.py]: even better with:

if ex_expunge:

Comment (boul) [inline libcloud/compute/drivers/cloudstack.py]: Less is more :) Done.

**Pull request closed without merging**